spring-integration-core/src/main/java/org/springframework/integration/support/management/graph/IntegrationNode.java

```diff
@@ -47,8 +47,11 @@ protected IntegrationNode(int nodeId, String name, Object nodeObject, Stats stat
 		this.componentType = nodeObject instanceof NamedComponent ? ((NamedComponent) nodeObject).getComponentType()
 				: nodeObject.getClass().getSimpleName();
 		this.stats = stats;
-		if (nodeObject instanceof ExpressionBased) {
-			this.properties.put("expression", ((ExpressionBased) nodeObject).getExpressionString());
+		if (nodeObject instanceof IntegrationObjectSupport) {
```

Comment:
Yeah... The problem is here that `IOS` is `class`, so proxy issue has a place anyway.
I agree that the hell with all those marker interfaces wearies a bit, but that not so bad during analysis to distinguish components by their marker interfaces.
I don't mind for `ExpressionBased`, I'm just not thrilled the package for it...

You arguments to do that in the `IOS`?